include/AST/Stmt/CompoundStmt.hpp

```diff
@@ -13,50 +15,68 @@ namespace glu::ast {
 ///
 /// This class inherits from StmtBase and encapsulates the details of a compound
 /// statement, including its list of statements.
-class CompoundStmt : public StmtBase {
-    llvm::SmallVector<StmtBase *> _stmts;
+class CompoundStmt final
+    : public StmtBase,
+      private llvm::TrailingObjects<CompoundStmt, StmtBase *> {
+    using TrailingArgs = llvm::TrailingObjects<CompoundStmt, StmtBase *>;
+    friend TrailingArgs;
+
+    unsigned _stmtCount;
+
+    // Method required by llvm::TrailingObjects to determine the number
+    // of trailing objects.
+    size_t
+        numTrailingObjects(typename TrailingArgs::OverloadToken<StmtBase *>) const
+    {
+        return _stmtCount;
+    }
 
-public:
     /// @brief Constructor for the CompoundStmt class.
     /// @param location The source location of the compound statement.
     /// @param stmts A vector of StmtBase pointers representing the statements
     /// in the compound statement.
     CompoundStmt(SourceLocation location, llvm::SmallVector<StmtBase *> stmts)
         : StmtBase(NodeKind::CompoundStmtKind, location)
-        , _stmts(std::move(stmts))
+        , _stmtCount(stmts.size())
     {
-        for (auto stmt : _stmts)
+        for (auto &stmt : stmts)
```

Comment:
for (auto *stmt
or
for (StmtBase *stmt

they're already pointers, no need for extra indirection